### ISSUE TITLE
registry module does not enforce type

### ISSUE THREAD

**Issue by doesitblend** [saltstack/salt#44192]
### Description of Issue/Question
When you pass a unicode string to the set_value function with a type of REG_DWORD it will fail because it does not get converted to an INT.

### Setup
```
router_gpo_1: 
  lgpo.set: 
    - computer_policy: 
        "Domain controller: Allow server operators to schedule tasks": Disabled 
```

### Steps to Reproduce Issue
Running the above state will provide the following error:

```
windows_host:
----------
          ID: lgpo_test
    Function: lgpo.set
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "c:\salt\bin\lib\site-packages\salt\state.py", line 1765, in call
                  **cdata['kwargs'])
                File "c:\salt\bin\lib\site-packages\salt\loader.py", line 1705, in wrapper
                  return f(*args, **kwargs)
                File "c:\salt\bin\lib\site-packages\salt\states\win_lgpo.py", line 294, in set_
                  adml_language=adml_language)
                File "c:\salt\bin\lib\site-packages\salt\modules\win_lgpo.py", line 5348, in set_
                  raise CommandExecutionError(msg.format(regedit))
              CommandExecutionError: Error while attempting to set policy Domain controller: Allow server operators to schedule tasks via the registry.  Some changes may not be applied as expected
     Started: 15:33:50.693000
    Duration: 1578.0 ms
     Changes:   

Summary for windoze
------------
Succeeded: 0
Failed:    1
```

This will produce the following traceback on the minion (debug code added into reg.py to produce traceback message):

```
Traceback (most recent call last):
  File "c:\salt-dev\salt\salt\modules\reg.py", line 604, in set_value
    RegSetValueEx(handle, local_vname, 0, vtype_value, local_vdata)
SystemError: ..\Objects\longobject.c:426: bad argument to internal function
```
### Versions Report
Salt 2017.7.2 
Python 2.7.13
Windows Server 2012 R2

**Comment by doesitblend** [comment]
@twangboy Any comment on this?

**Comment by twangboy** [comment]
The PR above will go into the 2016.11 branch and be merged forward to 2017.7. It has the `2016.11.9` and `2017.7.3` tags on it. This will fix the issue where the LGPO module is sending a Unicode value where the reg module is expecting an integer value.
